packages/state_queue_test/lib/src/state_queue_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'package:flutter/widgets.dart';
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+import 'package:state_queue/state_queue.dart';
+import 'package:test/test.dart' as test;
+
+@isTest
+void blocTest<B extends StateQueue<State>, State>(
+  String description, {
+  @required B Function() build,
+  FutureOr<void> Function(B bloc) act,
+  Iterable<State> expect,
+}) {
+  test.test(description, () async {
+    await runBlocTest<B, State>(
+      build: build,
+      act: act,
+      expect: expect,
+    );
+  });
+}
+
+@visibleForTesting
+Future<void> runBlocTest<B extends StateQueue<State>, State>({
+  @required B Function() build,
+  FutureOr<void> Function(B bloc) act,
+  Iterable<State> expect,
+}) async {
+  final collectStates = expect != null;
+
+  final states = <State>[];
+
+  final bloc = build();
+
+  void collectState() {
+    states.add(bloc.value);
+  }
+
+  if (collectStates) {
+    bloc.addListener(collectState);
+  }
+
+  if (collectStates) {
+    states.add(bloc.value);
```

Comment:
Interesting. I would also think it makes sense to test the initial state if there are any parameters to it. Else one would have to setup the bloc twice with the same data. Not sure if that is not too granular.

I think this is what `skip` is for, which for some kinds of blocs (which boring / static loading states), might just be the thing to do (though personally I think I would rather assert on the loading instead of skipping it, both of which are 1 line).

For reference here's `bloc_test` (which also uses `setUp` to "build" the bloc instead of doing it inline):

![image](https://user-images.githubusercontent.com/52437/93740977-d523c900-fbeb-11ea-808a-98dfb9391b86.png)


![image](https://user-images.githubusercontent.com/52437/93740962-cf2de800-fbeb-11ea-9a54-e9ad0d08f406.png)

Reply:
yes, this is what I meant. From this example, you can clearly see that when using the `blocTest` wrapper function, the initial state `0` is not emitted, only the changes that were made during `act`. I am fine with both approaches, so having the initial state always in the state recordings, or leaving it out. Just wanted to mention it. So let's leave the initial state in the recordings, yes?

Reply:
> So let's leave the initial state in the recordings, yes?

Agree :)